src/mz/src/context.rs

```diff
@@ -81,6 +82,62 @@ impl Context {
         })
     }
 
+    /// Retrieves the admin endpoint from the configuration file.
+    ///
+    /// - If an admin-endpoint is provided, it uses its value.
+    /// - If only a cloud-endpoint is provided, it constructs the admin endpoint based on it.
+    /// - If neither an admin-endpoint nor a cloud-endpoint is provided, default values are used.
+    pub fn get_admin_endpoint(
+        &self,
+        cloud_endpoint: Option<Url>,
+        admin_endpoint: Option<&str>,
+    ) -> Result<Option<Url>, ParseError> {
+        if let Some(admin_endpoint) = admin_endpoint {
+            return Ok(Some(admin_endpoint.parse()?));
+        } else if let Some(cloud_endpoint) = cloud_endpoint {
+            let mut admin_endpoint_url = cloud_endpoint;
+
+            if let Some(host) = admin_endpoint_url.host_str().as_mut() {
+                admin_endpoint_url.set_host(Some(&host.replace("api.", "admin.")))?;
```

Comment:
This'll replace "api." anywhere in the string, which is a bit too aggressive. You could instead do something like this to only replace "api." at the beginning of the string:

```suggestion
                if let Some(host) = host.strip_prefix("api.") {
                    admin_endpoint_url.set_host(format!("admin.{}", host));
                }
```